File: backend/internal/model/key.go
package model

import "time"

type Key struct {
	ID          int64     `json:"id"`
	KeyValue    string    `json:"key_value" binding:"required"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateKeyRequest struct {
	KeyValue    string `json:"key_value" binding:"required"`
	Description string `json:"description"`
}

type UpdateKeyRequest struct {
	KeyValue    string `json:"key_value,omitempty"`
	Description string `json:"description,omitempty"`
}

File: frontend/src/commonMain/kotlin/com/transportpay/data/models/Key.kt
package com.transportpay.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Key(
    val id: Int,
    @SerialName("key_value") val keyValue: String,
    val description: String?,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String
)

@Serializable
data class CreateKeyRequest(
    @SerialName("key_value") val keyValue: String,
    val description: String? = null
)

@Serializable
data class UpdateKeyRequest(
    @SerialName("key_value") val keyValue: String? = null,
    val description: String? = null
)
